narwhals/_ibis/namespace.py

```diff
@@ -0,0 +1,268 @@
+        def func(df: IbisLazyFrame) -> list[ir.Value]:
+            cols = chain.from_iterable(expr(df) for expr in exprs)
+            return [reduce(operator.or_, cols)]
+
+        return self._expr(
+            call=func,
+            evaluate_output_names=combine_evaluate_output_names(*exprs),
+            alias_output_names=combine_alias_output_names(*exprs),
+            backend_version=self._backend_version,
+            version=self._version,
+        )
+
+    def max_horizontal(self, *exprs: IbisExpr) -> IbisExpr:
+        def func(df: IbisLazyFrame) -> list[ir.Value]:
+            cols = list(chain.from_iterable(expr(df) for expr in exprs))
+            return [ibis.greatest(*cols).name(cols[0].get_name())]
+
+        return self._expr(
+            call=func,
+            evaluate_output_names=combine_evaluate_output_names(*exprs),
+            alias_output_names=combine_alias_output_names(*exprs),
+            backend_version=self._backend_version,
+            version=self._version,
+        )
+
+    def min_horizontal(self, *exprs: IbisExpr) -> IbisExpr:
+        def func(df: IbisLazyFrame) -> list[ir.Value]:
+            cols = list(chain.from_iterable(expr(df) for expr in exprs))
+            return [ibis.least(*cols).name(cols[0].get_name())]
+
+        return self._expr(
+            call=func,
+            evaluate_output_names=combine_evaluate_output_names(*exprs),
+            alias_output_names=combine_alias_output_names(*exprs),
+            backend_version=self._backend_version,
+            version=self._version,
+        )
+
+    def sum_horizontal(self, *exprs: IbisExpr) -> IbisExpr:
+        def func(df: IbisLazyFrame) -> list[ir.Value]:
+            cols = [
+                e.fill_null(lit(0)).name(e.get_name())
+                for _expr in exprs
+                for e in _expr(df)
+            ]
+            return [reduce(operator.add, cols)]
+
+        return self._expr(
+            call=func,
+            evaluate_output_names=combine_evaluate_output_names(*exprs),
+            alias_output_names=combine_alias_output_names(*exprs),
+            backend_version=self._backend_version,
+            version=self._version,
+        )
+
+    def mean_horizontal(self, *exprs: IbisExpr) -> IbisExpr:
+        def func(df: IbisLazyFrame) -> list[ir.Value]:
+            expr = (
+                cast("ir.NumericColumn", e.fill_null(lit(0)))
+                for _expr in exprs
+                for e in _expr(df)
+            )
+            non_null = (
+                cast("ir.NumericColumn", e.isnull().ifelse(lit(0), lit(1)))
+                for _expr in exprs
+                for e in _expr(df)
+            )
+            first_expr_name = exprs[0](df)[0].get_name()
+
+            def _name_preserving_sum(
+                e1: ir.NumericColumn, e2: ir.NumericColumn
+            ) -> ir.NumericColumn:
+                return cast("ir.NumericColumn", (e1 + e2).name(e1.get_name()))
```

Comment:
these name-preserving operations shouldn't be necessary, as we set the name as late as possible anyway

if this was taken from the dask implementation, it's probably something we forgot to clean up - i'll remove it from there in https://github.com/narwhals-dev/narwhals/pull/2473

Reply:
Thanks, cleaned up here for Ibis too